src/Algebra/Morphism/Bundles.agda

```diff
@@ -0,0 +1,241 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Bundled definitions of homomorphisms between algebras
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Algebra.Morphism.Bundles where
+
+open import Algebra.Bundles -- using (Magma)
+open import Algebra.Morphism.Structures -- using (IsMagmaHomomorphism)
+open import Level using (Level; suc; _⊔_)
+open import Relation.Binary.Morphism using (IsRelHomomorphism)
+open import Relation.Binary.Morphism.Bundles using (SetoidHomomorphism)
+
+private
+  variable
+    ℓ a ℓa b ℓb : Level
+
+
+------------------------------------------------------------------------
+-- Morphisms between Magmas
+------------------------------------------------------------------------
+
+record MagmaHomomorphism (A : Magma a ℓa) (B : Magma b ℓb) : Set (a ⊔ b ⊔ ℓa ⊔ ℓb) where
```

Comment:
Could you add a brief comment documenting why you decided to use `Bundles` here instead of the `RawBundles` used by the morphism structures? Having thought about it for all of a minute, it's not terribly obvious to me why we should make the switch. In particular, these morphisms are often used to transfer properties from one structure, e.g. a magma, to prove that another structure is also a magma. Assuming that they're both a magma to starts with, prevents you from using these morphisms for that purpose.

Reply:
Oh-ho-ho! Good catch... will have to go back to the drawing board and scratch my head for a bit. 

My first (worst?) thought is that such a use case could be handled via the already-existing `IsMagmaHomomorphism` infrastructure... but I have *not* tested that conjecture (related to @Taneb 's on `Monomorphism`s?)... while the specific PR (#1962 ) which gave rise to (the use case giving rise to) this PR *does* already have the two `Magma` bundles at hand... and I'd never reconsidered that state of affairs...

Reply:
So yes, it can (and is!) handled by `IsMagmaHomomorphism` and co. The question is what does making them `Bundles` instead of `RawBundles` actually gains us? I can only see downsides.

Reply:
I think I'm convinced: but there's a knock-on consequence... : we can no longer define a `SetoidHomomorphism` manifest field at the bottom of the hierarchy... and probably `Relation.Morphism.Bundles` should be refactored to reflect (the eventual introduction of) `RawSetoid` as index for `SetoidHomomorphism`... ;-) cf. your #1464 / commit 981602398f183ca901a32208673e2fcf1f20caac

Reply:
More knock-on:
* For `Algebra.Morphism.Construct.Identity`, we need `refl` for `IsXHomomorphism`, so do we parametrise by `xHomomorphism` by the full `X` bundle, or the `RawX` bundle, with an explicit `refl` witness...?
* Ditto. with `trans` for  `Algebra.Morphism.Construct.Composition`...

Advice welcome!

Reply:
Latest commits: 
* `RawX` bundle for the *definition*
* `X` bundle for the `Construct`ions